src/get-started/flutter-for/ios-devs.md

```diff
@@ -490,200 +859,593 @@ class SignaturePainter extends CustomPainter {
+  </li>
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.repo.this}}/{{sample_path}}/lib/navigation.dart">View on Github</a>
+  </li>
+</ul>
+</nav>
+
+<?code-excerpt "lib/navigation.dart (DetailsPage)"?>
 ```dart
-@override
-Widget build(BuildContext context) {
-  return const Center(
-    child: CustomButton('Hello'),
-  );
+class DetailsPage extends StatelessWidget {
+  const DetailsPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    // read the person instance from the arguments
+    final Person person = ModalRoute.of(
+      context,
+    )?.settings.arguments as Person;
+    // extract the age
+    final age = '${person.age} years old';
+    return Scaffold(
+      // display name and age
+      body: Column(children: [Text(person.name), Text(age)]),
+    );
+  }
 }
 ```
 
-## Navigation
+For more advanced navigation and routing requirements, 
+you should use a routing package such as [go_router][]. 
+You can find more details at Navigation in Flutter.  
 
-### How do I navigate between pages?
+### Manually pop back
 
-In iOS, to travel between view controllers, you can use a
-`UINavigationController` that manages the stack of view controllers
-to display.
+In SwiftUI, for situations where your view needs 
+to perform a manual pop-back to the previous screen, 
+you use the `dismiss` environment value as follows:
 
-Flutter has a similar implementation,
-using a `Navigator` and `Routes`.
-A `Route` is an abstraction for a “screen” or “page” of an app,
-and a `Navigator` is a [widget][]
-that manages routes. A route roughly maps to a
-`UIViewController`. The navigator works in a similar way to the iOS
-`UINavigationController`, in that it can `push()` and `pop()`
-routes depending on whether you want to navigate to, or back from, a view.
+```swift
+Button("Pop back") {
+        dismiss()
+      }
+```
 
-To navigate between pages, you have a couple options:
+In Flutter, to achieve the same effect, you use the `pop()` 
+function of the `Navigator` class as follows:
+
+<nav class="navbar bg-primary">
+ <ul class="navbar-nav navbar-code ml-auto">
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.dartpad}}/?id=0cf352feaeaea2eb107f784d879e480d">Test in DartPad</a>
+  </li>
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.repo.this}}/{{sample_path}}/lib/popback.dart">View on Github</a>
+  </li>
+</ul>
+</nav>
+
+<?code-excerpt "lib/popback.dart (PopBackExample)"?>
+```dart
+TextButton(
+  onPressed: () {
+    // this is the main code that allows the
+    // view to pop back to its presenter
+    Navigator.of(context).pop();
+  },
+  child: const Text('Pop back'),
+),
+```
 
-* Specify a `Map` of route names.
-* Directly navigate to a route.
+### Navigating to another app
 
-The following example builds a `Map.`
+In SwiftUI, to open a URL to another application, you use the `openURL` environment 
+variable, as follows:
```

Comment:
Ditto. It's opening a website page, not an app.

Reply:
Changed the text